ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
+
+    public
 
     def initialize(source_or_path_or_io)
       @line_number = 0
+
       if String === source_or_path_or_io
         if File.file?(source_or_path_or_io)
           @io = File.open(source_or_path_or_io, 'r:BOM|UTF-8')
         else
           @io = StringIO.new(source_or_path_or_io)
         end
+      elsif TokenScanner === source_or_path_or_io
```

Comment:
Shouldn't be allowed to pass a scanner to a scanner's ctor. See how it's done in js/python/java